pkgs/native_assets_cli/lib/src/validator/validator.dart

```diff
@@ -0,0 +1,207 @@
+// Copyright (c) 2024, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'dart:io';
+
+import '../api/asset.dart';
+import '../api/build_config.dart';
+import '../api/build_output.dart';
+import '../api/hook_config.dart';
+import '../api/link_config.dart';
+import '../api/link_mode_preference.dart';
+
+typedef ValidateResult = ({
+  bool success,
+  List<String> errors,
+});
+
+// TODO(dacoharkes): More validation in the native_assets_builder.
+// TODO(dacoharkes): NativeCodeAssets and DataAssets should have a file if not
+// dry run.
+Future<ValidateResult> validateBuild(
```

Comment:
nit: Could these functions be annotated with `@visibleForTesting`? Applicable for all the functions exported from `*_internal.dart`